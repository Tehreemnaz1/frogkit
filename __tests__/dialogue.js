import { mount, shallow } from 'vue-test-utils'
import Vue from 'vue'
import Dialogue from '@/components/dialogue'

const propsData = {
  visible: true,
  dark: true,
  requiredAction: false,
  lang: 'de',
  title: 'Hiergeblieben!',
  text: 'Smartfrog schenkt dir den ersten Monat!',
  codeText: 'Gutscheincode:',
  code: 'FREE1',
  btnText: 'Jetzt einlÃ¶sen',
  noText: 'Jetzt nicht'
}

describe('Dialogue component', () => {
  it('should render component and match snapshot', () => {
    const wrapper = mount(Dialogue, { propsData })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Should emit usePromo event when onUsePromo method is called', () => {
    const cmp = shallow(Dialogue, { propsData })
    const stub = jest.fn()
    cmp.vm.$on('usePromo', stub)
    cmp.vm.onUsePromo()
    expect(stub).toBeCalled()
  })

  it('Should emit closed event when close method is called', () => {
    const cmp = shallow(Dialogue, { propsData })
    const stub = jest.fn()
    cmp.vm.$on('closed', stub)
    cmp.vm.close()
    expect(stub).toBeCalled()
  })
})
