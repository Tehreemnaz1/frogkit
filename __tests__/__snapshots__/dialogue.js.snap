// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialogue component should render component and match snapshot 1`] = `
<flex justify="center" align="center" class="fk-dialogue fk-dialogue--dark v-enter fk-dialogue__animation-in">
    <section class="fk-dialogue__frame">
        <a class="fk-dialogue__close">
            <icon icon="close-light" color="orange"></icon>
        </a>
        <couponpopup lang="de" title="Hiergeblieben!" text="Smartfrog schenkt dir den ersten Monat!" codetext="Gutscheincode:" code="FREE1" btntext="Jetzt einlÃ¶sen" notext="Jetzt nicht"></couponpopup>
    </section>
</flex>
`;
